### PR TITLE
Add support for passing collapse-id to APNS

### PR DESCRIPTION
**Pull request by Amnell**
Added support for adding `apns-collapse-id` to enable collapsing of pushes just like the `apns-expiration`.

I changed the method signature for `APNS_generateNotification` from `APNS._generateNotification(data, expirationDate)` to `APNS._generateNotification(data, headers)` making it easier to add more header properties in the future.

# headers
The headers parameter is a JS object with the following supported properties:

- expirationTime
- collapseID
- topic

Don't hesitate to feedback on the code as JS isn't my "native" language :)

### CODE REVIEW

**Comment by flovilmart** [comment]
Thanks for the PR. Can you provide links to the docs from the APN lib as well as apple's so I can doublecheck if it's implemented as expected?

**Comment by Amnell** [comment]
@flovilmart 
[Apple Documentation 
(Table 8-2)](https://developer.apple.com/library/content/documentation/NetworkingInternet/Conceptual/RemoteNotificationsPG/CommunicatingwithAPNs.html)
[APN lib documentation](https://github.com/node-apn/node-apn/blob/a8422e0b5f4cc78002f33b680092223369d74861/doc/notification.markdown)
[APN source reference](https://github.com/node-apn/node-apn/blob/5d6bfb43c6540694e496cce136e3df54bb55dfa5/lib/notification/index.js#L58)

File: src/APNS.js [hunk 1]
```diff
@@ -95,7 +96,8 @@ export class APNS {
         continue;
       }
 
-      let notification = APNS._generateNotification(coreData, expirationTime, appIdentifier);
+      let headers = { expirationTime: expirationTime, appIdentifier: appIdentifier, collapseId: collapseId }
```
Comment (flovilmart) [inline src/APNS.js]: the appIdentifier header should be renamed  `topic`

**Comment by flovilmart** [comment]
Thanks for the docs! Very helpful :) Just one small nit.

**Comment by Amnell** [comment]
Great. Good spot :)

File: src/APNS.js (hunk 1 above)
Comment (Amnell) [inline src/APNS.js]: Great. Good spot 👍

**Comment by flovilmart** [comment]
Looking great! thanks!

**Pull request merged**